java/ql/test/stubs/springframework-5.2.3/org/springframework/security/web/savedrequest/SimpleSavedRequest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2004, 2005, 2006 Acegi Technology Pty Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web.savedrequest;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import javax.servlet.http.Cookie;
+
+import org.springframework.util.Assert;
+
+/**
+ * A Bean implementation of SavedRequest
+ *
+ * @author Rob Winch
+ * @since 5.1
+ */
+public class SimpleSavedRequest implements SavedRequest {
+
+	private String redirectUrl;
+
+	private List<Cookie> cookies = new ArrayList<>();
+
+	private String method = "GET";
+
+	private Map<String, List<String>> headers = new HashMap<>();
+
+	private List<Locale> locales = new ArrayList<>();
+
+	private Map<String, String[]> parameters = new HashMap<>();
+
+	public SimpleSavedRequest() {
+	}
+
+	public SimpleSavedRequest(String redirectUrl) {
+		this.redirectUrl = redirectUrl;
+	}
+
+	public SimpleSavedRequest(SavedRequest request) {
+		this.redirectUrl = request.getRedirectUrl();
+		this.cookies = request.getCookies();
+		for (String headerName : request.getHeaderNames()) {
+			this.headers.put(headerName, request.getHeaderValues(headerName));
+		}
+		this.locales = request.getLocales();
+		this.parameters = request.getParameterMap();
+		this.method = request.getMethod();
```

Comment:
```suggestion
```
Could you remove all the method bodies? (replace with `return null;` where applicable). We shouldn't really have actual code in the stubs.